### ISSUE TITLE
[Bug]: Being able to persist my default model which is not the OpenAI key (fabric default) 

### ISSUE THREAD

**Issue by QCadjunct** [danielmiessler/fabric#767]
### What happened?

[Bug]: Fabric Dependency Issues with gRPC and Google Cloud Libraries] https://github.com/danielmiessler/fabric/issues/747#top)
https://github.com/danielmiessler/fabric/issues/747

## There needs to be a way to persist the DefaultModel in fabric --changeDefaultModel based upon the --changeDefaultModel command.

I only want to use Google's models/gemini-1.5-pro and ollama models. **I want exclude for the time being OpenAI (fabric default) and Claude.ai.**
##  In the re-installation process
1.fabric --listmodels and found models/gemini-1.5-pro
1.fabric --changeDefaultModel models/gemini-1.5-pro

###At this point the gRPC error, was still occurring but I was getting answers from fabric
Once, I reboot OpenAI took precedence and returned the error "Error: Client error '401 Unauthorized' for url 'https://api.openai.com/v1/models'". This reminds me of the "Harry Belafonte & Odetta - A Hole in the Bucket: "https://www.youtube.com/watch?v=xVAvMIhvqfk"

Also, add as many keys as you like but have a way to reset a combination of keys which are dynamically persisted but can be changed and restart the shell and happy fabric-ing to handle as an example (Fabric: This OPENSOURCE AI Framework can AUTOMATE YOUR LIFE & ANY TASK (Setup with Ollama & Groq) : "https://www.youtube.com/watch?v=Iup3-Jvkrhg" ) as a permutation of the ability to reset a combination of keys

## Reboot error : I followed the suggested format a similar way described by Joshua Fuller

After a reboot, I get this error "Error: Client error '401 Unauthorized' for url 'https://api.openai.com/v1/models'
For more information check: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/401"

### Full Uninstall and Reinstallation:

Uninstalled fabric using pipx: pipx uninstall fabric

1. Deleted the local fabric repository: rm -rf ~/github/fabric
2. Re-cloned the fabric repository from GitHub: git clone https://github.com/danielmiessler/fabric.git
3. Reinstalled fabric using pipx: pipx install fabric
4. Re-setup fabric as a new installation: fabric --setup
5. Bypassed the openai key and claude.ai key
6. Inserted Google generative ai api key and youtube api key
7. I performed these steps twice to ensure a clean installation.
8. I tested that fabric was installed properly:

### Version check

- [X] Yes I was.

### Relevant log output

```shell
After a reboot, I get this error "Error: Client error '401 Unauthorized' for url 'https://api.openai.com/v1/models'
For more information check: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/401"

It requires me to use the temporary suggestion in https://github.com/danielmiessler/fabric/issues/747 to get it to work temporarily.  Forcing me to reinstall after any reason to reboot my machines.
```

**Comment by matigumma** [comment]
its right....
api call to api.openai.com/v1/models require an api_key....

![image](https://github.com/user-attachments/assets/e19513d1-bd6b-4c5a-a202-c2c780db2c0a)


by default appear to use gpt-4-turbo-preview as DEFAULT_MODEL

![image](https://github.com/user-attachments/assets/c5359f2c-5649-4154-8158-be7f63dba6c2)

and line 417 on utils asume it is configured and try to call openai service

![image](https://github.com/user-attachments/assets/8da9712b-cbff-45b7-837f-390341cc09aa)

**Comment by QCadjunct** [comment]
1.  I believe on my last fabric -setup, I hit enter for OpenAI and Claude.ai.  I did enter the same key for generative ai key for Google and YouTube key.
  2.  After the fabric setup completed, I did a fabric -listmodels
  3.  I found the Google models/gemini-1.5-pro and did a fabric -changeDefaultModel models/gemini-1.5-pro
  4.  Now, tow OS reboots later, it is functioning without OpenAI and Claude.ai

Allow for as many keys to loaded, but have a pair of persisted Keys that can be persisted and changed by restarting the shell.

Finally, my other is screwed up fab was installed but not fabric.  How remove safer the python environment for a fresh install of everything?

Thanks for getting back to me,

Peter
